### PR TITLE
added "on_left_side" property for bots

### PR DESCRIPTION
**Pull request by sappelhoff**
fix issue #49

### CODE REVIEW

**Comment by Debilski** [comment]
My personal take: This should at most be a property on `AbstractPlayer` but not in the datamodel API. But I have always been against this suggestion by @pberkes anyway (I think no-one should ever need it) so I am not the person to decide this one. :)

**Comment by pberkes** [comment]
Hi @sappelhoff, thanks for the contribution! I needed this while writing my own bots.

Rike, can you clarify why you don't want this method, or where it should go? If there is a better way, can you please add a comment to issue #49 and close it?

File: pelita/datamodel.py [hunk 1]
```diff
@@ -115,6 +115,13 @@ def is_destroyer(self):
         return self.in_own_zone
 
     @property
+    def on_left_side(self):
+        # True if current_pos is left of team_border
+        return (self.current_pos[0] < ((sorted(self.homezone)[1] + 1)
```
Comment (Debilski) [inline pelita/datamodel.py]: `sorted(self.homezone)[1]` is equivalent to `max(self.homezone)`, right?

**Comment by Debilski** [comment]
@pberkes I find it too specific (and it kind of relies on how we set up the initial rules). But yeah, the rules are not very likely to change.

@sappelhoff can you rename it to `on_west_side` or similar. And please add a test.

File: pelita/datamodel.py (hunk 1 above)
Comment (Debilski) [reply pelita/datamodel.py]: And that nested if is a little convoluted. I think a verbose multi-line if is better here.

Comment (sappelhoff) [reply pelita/datamodel.py]: @Debilski, I thought I would have to import math or numpy for a max function ... just realized now that it's not necessary to import anything for that :-) 

Next commit will be more verbose

**Comment by Debilski** [comment]
> I have some trouble adding a test.
> 
> When adding the following to test/test_datamodel.py:
> 
> def test_on_west_side(self):
>     bot = Bot(1, (6, 6), 1, (3, 6), current_pos = (1, 1))
>     assert bot.on_west_side
> I get this error upon running pytest test/test_datamodel.py:
> 
> E AttributeError: 'Bot' object has no attribute 'on_west_side'
> 
> Furthermore, when I go to datamodel.py and change e.g., the is_harvester method to is_harvester2, the tests still pass despite the fact that the is_harvester method should be missing ...
> 
> Any idea why this happens @Debilski?

Hi, @sappelhoff, I think this is because you still run the tests against the old pelita. You have to install the dev version from the local git repo with the `-e` flag:

    cd pelita
    pip install -e .

**Comment by sappelhoff** [comment]
> cd pelita
pip install -e .


Thanks Rike, that was it 👍 

I made the changes and opened a new pull request with a clean commit history and proper branch name. See pull request #430. Please just close this one, I kind of messed up rebasing/squashing the commits and don't know what happened.

**Comment by pberkes** [comment]
@Debilski did you want @sappelhoff to move the code to `AbstractPlayer`?

**Comment by Debilski** [comment]
@pberkes On second thought it only makes sense to have it on `AbstractPlayer` if you are only going to use it with your current players and never on the other bots (which may be exactly the use case but I am not sure).

I am open to postpone this decision for a more general API restructuring.

**Comment by Debilski** [comment]
Closed in favour of #430

**Pull request closed without merging**